flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/AbstractFileStoreScan.java

```diff
@@ -206,7 +206,22 @@ public Plan plan() {
                             "Unknown value kind " + entry.kind().name());
             }
         }
-        List<ManifestEntry> files = new ArrayList<>(map.values());
+        List<ManifestEntry> files = new ArrayList<>();
+        for (ManifestEntry file : map.values()) {
+            if (checkNumOfBuckets) {
+                Preconditions.checkState(
+                        file.totalBuckets() == numOfBuckets,
+                        "Trying to add file %s "
```

Comment:
Actually there is no corrupted, can we enrich this error message? For example, remind the user that if you need to write this old bucket number of partitions, you need to manually rescale it to do so.